### ISSUE TITLE
Frequent uwsgi container crashes when filtering in UI

### ISSUE THREAD

**Issue by franco-cucchi** [DefectDojo/django-DefectDojo#6901]
**Bug description**
It seems that applying filters on the UI most of the times causes the nginx to report in the browser:

 "Sorry, the page you are looking for is currently unavailable.
Please try again later.

If you are the system administrator of this resource then you should check the error log for details.

Faithfully yours, nginx."  

and wsgi container to crash and restart.

**Steps to reproduce**
1. Go to 'Simple Metrics'
2. Click on 'View details' for a Product Type
3. Click on filter icon to expand filters
4. Enter some filters (i.e. start date, end date, product type, active-no, closed-yes)
5. Click on Apply Filters

The same happens even when starting from the findings and filtering the search.

**Expected behavior**
The UI should report the correct report based on applied filters

**Deployment method** *(select with an `X`)*
- [ ] Docker Compose
- [X ] Kubernetes
- [ ] GoDojo

**Environment information**
 - Operating System: Container-optimised OS with containerd (Google Cloud Platform - GKE)
 - DefectDojo version 2.14.3
 - CloudSQL - postgres 13
 - redis deployed as pod in GKE

**Logs** 

     "[pid: 1|app: -|req: -/-] 10.108.2.1 (-) {30 vars in 415 bytes} [Mon Sep 26 09:03:27 2022] GET /login?force_login_form&next=/ => generated 18419 bytes in 89 msecs (HTTP/1.1 200) 9 headers in 480 bytes (2 switches on core 2)"
     "[26/Sep/2022 09:03:28] DEBUG [dojo.forms:2121] templates: [('', '---'), ('issue-trackers/jira_full', 'jira_full'), ('issue-trackers/jira_limited', 'jira_limited')]"
     "[26/Sep/2022 09:03:54] INFO [dojo.models:4056] enabling audit logging"
     "[26/Sep/2022 09:03:55] DEBUG [dojo.forms:2121] templates: [('', '---'), ('issue-trackers/jira_full', 'jira_full'), ('issue-trackers/jira_limited', 'jira_limited')]"
     "System check identified no issues (0 silenced)."
     "*** Starting uWSGI 2.0.20 (64bit) on [Mon Sep 26 09:03:56 2022] ***"
     "compiled with version: 10.2.1 20210110 on 22 September 2022 01:21:50"
     "os: Linux-5.10.127+ #1 SMP Fri Jul 8 00:39:24 UTC 2022"
     "nodename: defectdojo-django-7f5c9db89c-nwk8x"
     "machine: x86_64"
     "clock source: unix"
     "detected number of CPU cores: 8"
     "current working directory: /app"
     "detected binary path: /usr/local/bin/uwsgi"
     "!!! no internal routing support, rebuild with pcre support !!!"
     "*** WARNING: you are running uWSGI without its master process manager ***"
     "your memory page size is 4096 bytes"
     "detected max file descriptor number: 1048576"
     "lock engine: pthread robust mutexes"
     "thunder lock: disabled (you can enable it with --thunder-lock)"
     "uWSGI http bound on 0.0.0.0:8081 fd 3"
     "spawned uWSGI http 1 (pid: 22)"
     "uwsgi socket 0 bound to UNIX address /run/defectdojo/uwsgi.sock fd 6"
     "Python version: 3.8.13 (default, May 28 2022, 14:30:17)  [GCC 10.2.1 20210110]"
     "Python main interpreter initialized at 0x5594900a9f60"
     "python threads support enabled"
     "your server socket listen backlog is limited to 100 connections"
     "your mercy for graceful operations on workers is 60 seconds"
     "mapped 2079840 bytes (2031 KB) for 100 cores"
     "*** Operational MODE: preforking+threaded ***"
     "[26/Sep/2022 09:03:58] INFO [dojo.models:4056] enabling audit logging"
     "WSGI app 0 (mountpoint='') ready in 2 seconds on interpreter 0x5594900a9f60 pid: 1 (default app)"
     "*** uWSGI is running in multiple interpreter mode ***"
     "spawned uWSGI worker 1 (pid: 1, cores: 10)"
     "spawned uWSGI worker 2 (pid: 51, cores: 10)"
     "spawned uWSGI worker 3 (pid: 61, cores: 10)"
     "spawned uWSGI worker 4 (pid: 73, cores: 10)"
     "spawned uWSGI worker 5 (pid: 90, cores: 10)"
     "spawned uWSGI worker 6 (pid: 103, cores: 10)"
     "spawned uWSGI worker 7 (pid: 117, cores: 10)"
     "spawned uWSGI worker 8 (pid: 132, cores: 10)"
     "spawned uWSGI worker 9 (pid: 145, cores: 10)"
     "spawned uWSGI worker 10 (pid: 159, cores: 10)"
     "[26/Sep/2022 09:03:59] DEBUG [dojo.forms:2121] templates: [('', '---'), ('issue-trackers/jira_full', 'jira_full'), ('issue-trackers/jira_limited', 'jira_limited')]"
     "[pid: 51|app: -|req: -/-] 10.108.2.1 (-) {34 vars in 417 bytes} [Mon Sep 26 09:03:58 2022] GET /uwsgi_health => generated 18419 bytes in 2569 msecs (HTTP/1.1 200) 9 headers in 480 bytes (1 switches on core 0)"
     "[26/Sep/2022 09:04:04] DEBUG [dojo.forms:2121] templates: [('', '---'), ('issue-trackers/jira_full', 'jira_full'), ('issue-trackers/jira_limited', 'jira_limited')]"
     "[26/Sep/2022 09:04:04] DEBUG [dojo.forms:2121] templates: [('', '---'), ('issue-trackers/jira_full', 'jira_full'), ('issue-trackers/jira_limited', 'jira_limited')]"

When search is not crashing the subsequent log may be found:

"[pid: 1|app: -|req: -/-] 10.108.2.1 (-) {30 vars in 415 bytes} [Mon Sep 26 10:52:17 2022] GET /login?force_login_form&next=/ => generated 18419 bytes in 535 msecs (HTTP/1.1 200) 9 headers in 480 bytes (1 switches on core 3)"
 "[pid: 1|app: -|req: -/-] XX.XX.XX.XX (admin) {66 vars in 1497 bytes} [Mon Sep 26 10:52:04 2022] GET /metrics/product/type/18 => generated 2068843 bytes in 14260 msecs (HTTP/1.1 200) 9 headers in 405 bytes (10 switches on core 6)"

but no report is shown in the UI except the usual simple dashboards.

**Additional context** (optional)
The filtering functionality doesn't seem to work in general except for some of the simple reports, in example:

1) Critical Product Metrics -> OK
2) Product Type Metrics -> uwsgi crashes
3) Product Type Counts -> OK
4) Simple Metrics -> OK
5) Simple Metrics with filters -> uwsgi crashes
6) Engineer Metrics -> OK

This exact same behavior was also observed in previous installation version 2.9.0 from which we upgraded to version 2.14.3

**Comment by franco-cucchi** [comment]
Anybody commenting on this?

**Comment by mtesauro** [comment]
My guess is that something is making the request take too long - I'd assume its a resource constraint issue specific to your environment and not in DefectDojo generally since the demo site works without this issue.  Plus, the broader community hasn't chimed in on your issue so it would seem to be something specific to your particular deployment.

That error message basically says that Nginx waited as long as it was configured to wait for a response from uwsgi (DefectDojo) and it took too long.

So, I'd start with increasing the timeout of Nginx or looking at the resources allocated to the pod running DefectDojo/uwsgi.

**Comment by franco-cucchi** [comment]
Thanks for the comment.
I've increased the connection timeout now for nginx:

keepalive_timeout 120;

and ensured CPU/memory is far below pod's limits but uwsgi keeps on crashing after few seconds I hit the search button.
The fact that it keeps on crashing on specific steps makes me think that could be related to limited compatibility with underneath OS and therefore I wonder whether anyone has similar deployment as the one I've detailed previously.
Any further comment is really appreciated.

Thanks
